src/utils/resolver.js

```diff
@@ -0,0 +1,36 @@
+import debug from './debug'
+import * as xpath from './xpath'
+
+/**
+ * Resolve element relative to root
+ *
+ * @param {Element}       root
+ * @param {{ path, id }}  data
+ * @param {boolean}       throwException
+ */
+export function resolveElement(root, data, throwException = false) {
+  let transformObject = null
+  let { path, id } = data
+
+  if (id) {
+    transformObject = root.querySelector(`[data-spirit-id="${id}"]`)
+  }
+
```

Comment:
When I initially used the tool, because the data-spirit-id element wasn't present on the page, an element that I wasn't targeting started to animate. This was because the xPath was considered instead. 

I'm not sure there'd be a situation where if the data-spirit-id was missing from the DOM, the xPath would target the correct element. 

A solution might be to ensure the JSON only contains a data id OR the xPath and then this could could be updated to an `else if`. 

What do you think?

Reply:
Semi colons are missing across the project. Out of curiosity, why is this?

Reply:
Yes, using xpath is not very reliable, but for many it does the trick. 

Right now.. it tries to resolve elements by data id, and if it was unable to resolve it falls back to xpath. Think the current logic is pretty much the same as your solution right?

Reply:
Semicolons.. an ongoing debate. I know a lot of code standards tells you to use it.

This is just my opinion, I've switched back and forth and eventually decided to leave it out as I find it boilerplate and it doesn't add any value, besides without semicolons it's easier for the eye as well.

Reply:
As someone new to the project, it looked broken to me that elements I wasn't targeting were animating. I would've expected nothing to animate if the id's weren't present on the page with an error in the console explaining why.

Good to know about the semi colon's, I wasn't aware they weren't mandatory :)

Reply:
I totally agree with you, so you're saying to enforce people using id's.. without it simple doesn't work right?

The nice thing using xpath, is that you can literally animate any webpage and select any element. Maybe I should add a warning in the console.. but in the app you'll see that the selector is red instead of green:

![image](https://user-images.githubusercontent.com/232559/33989561-3c0c9ce2-e0c8-11e7-877b-8ae99069b46d.png)

Reply:
Sorry, I don't think we should enforce users to use ID's but if they have selected an element with an ID, then the JSON file shouldn't contain the xPath for it (which in turns means it wouldn't fall back to it in the event the ID is missing). However if the element was captured using only xPath, then it should work as intended.

I absolutely love the idea of viewing any site and animating it, it's what made is super accessible for our designers and I think is a great USP.

The colouring for xPath and ID makes it look like ID is right whilst the xPath is wrong. Maybe they could be the same style with different headers? xPath and Spirit ID?

Reply:
Ideally, using an data id is more reliable and thus better.. therefore using xpath is coloured red and ids green.

Good point about leaving out the xpath if an id is set.. will put it on the backlog, thanks!

Reply:
No, thank you! :)